_unittest_solvers/test_26_emit.py

```diff
@@ -1182,81 +1181,62 @@ def test_license_session(self, add_app):
-            return [best_band, best_freq]
+        number_of_runs = 5
+
+        # Do a run to ensure the license log exists
+        do_run()
+
+        # Find the license log for this process
+        appdata_local_path = os.getenv('LOCALAPPDATA')
+        pid = os.getpid()
+        dot_ansys_directory = os.path.join(appdata_local_path, 'Temp\\.ansys')
+        
+        license_file_path = ''
+        with os.scandir(dot_ansys_directory) as dir:
+            for file in dir:
+                if file.name.startswith('ansyscl') and file.name.endswith('.log') and str(pid) in file.name:
+                    license_file_path = os.path.join(dot_ansys_directory, file.name)
+                    break
         
-        # Warmup in case something isn't solved
-        best_case_rx_ch = {}
-        for rx in [receivers[0]]:
-            best_case_rx_ch[rx] = get_best_rx_channel(results, rx)
+        assert license_file_path != ''
         
-        # Get the time without the license session
-        start = time.perf_counter()
-        best_case_rx_ch = {}
-        for rx in [receivers[0]]:
-            best_case_rx_ch[rx] = get_best_rx_channel(results, rx)
-        end = time.perf_counter()
-        noLicenseSessionTime = end - start
+        def count_license_actions(license_file_path):
+            # Count checkout/checkins in most recent license connection
+            checkouts = 0
+            checkins  = 0
+            with open(license_file_path, 'r') as license_file:
+                lines = license_file.read().strip().split('\n')
+                for line in lines:
+                    if 'NEW_CONNECTION' in line:
+                        checkouts = 0
+                        checkins = 0
+                    elif 'CHECKOUT' in line:
+                        checkouts += 1
+                    elif 'CHECKIN' in line:
+                        checkins += 1
+            return (checkouts, checkins)
         
-        # Get the time using the license session
-        start = time.perf_counter()
-        best_case_rx_ch = {}
+        start_checkouts, start_checkins = count_license_actions(license_file_path)
+
+        # Run without license session
+        for i in range(number_of_runs):
+            do_run()        
+        
+        # Run with license session
         with revision.get_license_session():
-            for rx in [receivers[0]]:
-                best_case_rx_ch[rx] = get_best_rx_channel(results, rx)
-        end = time.perf_counter()
-        licenseSessionTime = end - start
+            for i in range(number_of_runs):
+                do_run()
+        
+        end_checkouts, end_checkins = count_license_actions(license_file_path)
+
+        checkouts = end_checkouts - start_checkouts
+        checkins = end_checkins - start_checkins
 
-        assert (licenseSessionTime*2) < noLicenseSessionTime
+        assert (checkouts == number_of_runs+1) and checkouts == checkins
```

Comment:
@bryankaylor I see the do_run method is called `number_of_runs` times twice (two separate loops). The second loop is inside a license_session so it should only have one license checkout for that loop. However, there is also a do_run on line 1194. So wouldn't there be `number_of_runs + 2` license checkouts?

Reply:
@myoung301 - The do_run on line 1194 is there because, if you run this test in isolation, the license file for the process doesn't exist yet. So, to ensure the test works whether it's run in isolation or with all the other tests, I added this first do_run that just makes the license file exist if it doesn't already.

After that do_run on line 1194, we parse the file and record the initial checkouts/checkins. Then, we do `number_of_runs` runs with and without a license session, re-parse the file, and record the final chckouts/checkins. So, since that first do_run is before we record the initial counts, the difference in checkouts should still be number_of_runs+1.